cram/cram_encode.c

```diff
@@ -3165,8 +3165,11 @@ int cram_put_bam_seq(cram_fd *fd, bam_seq_t *b) {
          * container as-is and then start a fresh one in a different mode.
          */
         if (multi_seq == 0 && fd->multi_seq == 1 && fd->multi_seq_user == -1) {
+            // User selected auto-mode, we're currently using multi-seq, but
+            // have detected we don't need to.  Switch back to auto.
             fd->multi_seq = -1;
         } else if (multi_seq || fd->multi_seq == 1) {
+            // We either detected multi-seq needed or user explicitly asked for it.
```

Comment:
Should it be then: `else if (multi_seq || fd->multi_seq`**_user**` == 1)` ?

Reply:
A very good point.  I clearly confused myself with that comment.  I'll try and test this and figure out what I intended when I wrote it!

Reply:
Ok it's necessary, but it could be either `fd->multi_seq` or `fd->multi_seq_user` as they're the same thing in this condition.  It's basically there because `fd->multi_seq` could (perhaps should) be set to zero but we're soldiering on using multi-ref mode because that's what we've been told to do.

However I reworked it so that the `multi_seq` variable assignment itself checks `fd->multi_seq_user` which avoids the need for this extra OR clause at all.